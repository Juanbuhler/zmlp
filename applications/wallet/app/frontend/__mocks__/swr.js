import { createElement } from 'react'

const noop = () => () => {}

/**
 * <SWRConfig />
 */

export const SWRConfig = ({ children, ...rest }) =>
  createElement('SWRConfig', rest, children)

/**
 * useSWR
 */

let mockUseSWRResponse = {}

export const __setMockUseSWRResponse = data => {
  mockUseSWRResponse = { revalidate: noop, ...data }
}

const useSWR = () => {
  return mockUseSWRResponse
}

export default useSWR
