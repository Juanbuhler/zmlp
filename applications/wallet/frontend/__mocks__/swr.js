import { createElement } from 'react'

const noop = () => () => {}

/**
 * <SWRConfig />
 */

export const SWRConfig = ({ children, ...rest }) =>
  createElement('SWRConfig', rest, children)

/**
 * cache
 */

export const { cache } = jest.requireActual('swr')

/**
 * useSWR
 */

let mockUseSWRResponse = { revalidate: noop, mutate: noop }

export const __setMockUseSWRResponse = (data) => {
  mockUseSWRResponse = { revalidate: noop, mutate: noop, ...data }
}

const useSWR = () => {
  return mockUseSWRResponse
}

export default useSWR
