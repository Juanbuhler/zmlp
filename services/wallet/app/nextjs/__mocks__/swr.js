import { createElement } from 'react'

/**
 * <SWRConfig />
 */

export const SWRConfig = ({ children, ...rest }) =>
  createElement('SWRConfig', rest, children)

/**
 * useSWR
 */

let mockUseSWRResponse = {}

export const __setMockUseSWRResponse = data => {
  mockUseSWRResponse = data
}

const useSWR = () => {
  return mockUseSWRResponse
}

export default useSWR
