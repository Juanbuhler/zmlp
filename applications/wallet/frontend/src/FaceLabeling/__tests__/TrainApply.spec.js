import TestRenderer from 'react-test-renderer'

import FaceLabelingTrainApply from '../TrainApply'

const PROJECT_ID = '76917058-b147-4556-987a-0a0f11e46d9b'
const JOB_ID = '1a1d45af-7477-1396-ae57-a618e8efb91f'

describe('<FaceLabelingTrainApply />', () => {
  it('should render properly when there are no unapplied changes and no training job', () => {
    require('swr').__setMockUseSWRResponse({
      data: { unappliedChanges: false, jobId: '' },
    })

    const component = TestRenderer.create(
      <FaceLabelingTrainApply projectId={PROJECT_ID} />,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render properly when there are unapplied changes and a training job', () => {
    require('swr').__setMockUseSWRResponse({
      data: { unappliedChanges: true, jobId: JOB_ID },
    })

    const component = TestRenderer.create(
      <FaceLabelingTrainApply projectId={PROJECT_ID} />,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
