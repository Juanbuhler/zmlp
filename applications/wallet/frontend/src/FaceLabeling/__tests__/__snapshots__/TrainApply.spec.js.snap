// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FaceLabelingTrainApply /> should render properly when unapplied changes is false and there is a training job 1`] = `
.emotion-7 {
  padding: 16px;
  border-bottom: 1px solid #3b3b3b;
}

.emotion-0 {
  height: 16px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  margin-left: 8px;
}

.emotion-5:hover div,
.emotion-5:focus-within div {
  visibility: visible;
  opacity: 1;
  -webkit-transition: all 0.5s ease 0.25s;
  transition: all 0.5s ease 0.25s;
}

.emotion-5:hover button,
.emotion-5:focus-within button {
  color: #ffffff;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 0;
  padding: 0;
  background-color: transparent;
  color: #808080;
}

.emotion-4 {
  position: absolute;
  right: -10px;
  top: 28px;
  color: #191919;
  background-color: #ffffff;
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
  padding: 12px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 400px;
  visibility: hidden;
  opacity: 0;
  -webkit-transition: all 0.5s ease 0.25s;
  transition: all 0.5s ease 0.25s;
}

.emotion-4:hover {
  visibility: visible;
  opacity: 1;
}

.emotion-3:before {
  content: ' ';
  position: absolute;
  top: -6px;
  right: 14px;
  border-bottom: 6px solid #ffffff;
  border-left: 6px solid transparent;
  border-right: 6px solid transparent;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1,
.emotion-1:hover,
.emotion-1:visited {
  background-color: #73b61c;
}

.emotion-1:hover {
  background-color: #96c955;
}

.emotion-1[aria-disabled=true] {
  background-color: #808080;
}

<div
  className="emotion-7"
>
  <FaceLebalingMessage
    error=""
    jobId="1a1d45af-7477-1396-ae57-a618e8efb91f"
    previousJobId="1a1d45af-7477-1396-ae57-a618e8efb91f"
    projectId="76917058-b147-4556-987a-0a0f11e46d9b"
    setPreviousJobId={[Function]}
  />
  <span>
    Once a name has been added to a face, training can begin. Names can continue to be edited as needed.
  </span>
  <div
    className="emotion-0"
  />
  <div
    className="emotion-6"
  >
    <button
      className="emotion-1"
      onClick={[Function]}
      type="button"
    >
      Override Current Training & Re-apply
    </button>
    <div
      className="emotion-5"
    >
      <button
        aria-details="TrainingHelp"
        aria-label="Training Help"
        className="emotion-2"
        type="button"
      >
        <svg
          data-file-name="help.svg"
          height={20}
        />
      </button>
      <div
        className="emotion-4"
        id="TrainingHelp"
        role="tooltip"
      >
        <div
          className="emotion-3"
        />
        There are new labels that have not been used for training. Restart training with the latest labels.
      </div>
    </div>
  </div>
</div>
`;

exports[`<FaceLabelingTrainApply /> should render properly when unapplied changes is true and there is no training job 1`] = `
.emotion-7 {
  padding: 16px;
  border-bottom: 1px solid #3b3b3b;
}

.emotion-0 {
  height: 16px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: not-allowed;
  color: #ffffff;
  background-color: transparent;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1,
.emotion-1:hover,
.emotion-1:visited {
  background-color: #73b61c;
}

.emotion-1:hover {
  background-color: #96c955;
}

.emotion-1[aria-disabled=true] {
  background-color: #808080;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  margin-left: 8px;
}

.emotion-5:hover div,
.emotion-5:focus-within div {
  visibility: visible;
  opacity: 1;
  -webkit-transition: all 0.5s ease 0.25s;
  transition: all 0.5s ease 0.25s;
}

.emotion-5:hover button,
.emotion-5:focus-within button {
  color: #ffffff;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 0;
  padding: 0;
  background-color: transparent;
  color: #808080;
}

.emotion-4 {
  position: absolute;
  right: -10px;
  top: 28px;
  color: #191919;
  background-color: #ffffff;
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
  padding: 12px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 400px;
  visibility: hidden;
  opacity: 0;
  -webkit-transition: all 0.5s ease 0.25s;
  transition: all 0.5s ease 0.25s;
}

.emotion-4:hover {
  visibility: visible;
  opacity: 1;
}

.emotion-3:before {
  content: ' ';
  position: absolute;
  top: -6px;
  right: 14px;
  border-bottom: 6px solid #ffffff;
  border-left: 6px solid transparent;
  border-right: 6px solid transparent;
}

<div
  className="emotion-7"
>
  <FaceLebalingMessage
    error=""
    jobId=""
    previousJobId=""
    projectId="76917058-b147-4556-987a-0a0f11e46d9b"
    setPreviousJobId={[Function]}
  />
  <span>
    Once a name has been added to a face, training can begin. Names can continue to be edited as needed.
  </span>
  <div
    className="emotion-0"
  />
  <div
    className="emotion-6"
  >
    <button
      aria-disabled={true}
      className="emotion-1"
      onClick={[Function]}
      type="button"
    >
      Train & Apply
    </button>
    <div
      className="emotion-5"
    >
      <button
        aria-details="TrainingHelp"
        aria-label="Training Help"
        className="emotion-2"
        type="button"
      >
        <svg
          data-file-name="help.svg"
          height={20}
        />
      </button>
      <div
        className="emotion-4"
        id="TrainingHelp"
        role="tooltip"
      >
        <div
          className="emotion-3"
        />
        There are new labels. Click to train and apply.
      </div>
    </div>
  </div>
</div>
`;

exports[`<FaceLabelingTrainApply /> should render properly when unapplied changes is true and there is no training job 2`] = `
.emotion-7 {
  padding: 16px;
  border-bottom: 1px solid #3b3b3b;
}

.emotion-0 {
  height: 16px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: not-allowed;
  color: #ffffff;
  background-color: transparent;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1,
.emotion-1:hover,
.emotion-1:visited {
  background-color: #73b61c;
}

.emotion-1:hover {
  background-color: #96c955;
}

.emotion-1[aria-disabled=true] {
  background-color: #808080;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  margin-left: 8px;
}

.emotion-5:hover div,
.emotion-5:focus-within div {
  visibility: visible;
  opacity: 1;
  -webkit-transition: all 0.5s ease 0.25s;
  transition: all 0.5s ease 0.25s;
}

.emotion-5:hover button,
.emotion-5:focus-within button {
  color: #ffffff;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 0;
  padding: 0;
  background-color: transparent;
  color: #808080;
}

.emotion-4 {
  position: absolute;
  right: -10px;
  top: 28px;
  color: #191919;
  background-color: #ffffff;
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
  padding: 12px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 400px;
  visibility: hidden;
  opacity: 0;
  -webkit-transition: all 0.5s ease 0.25s;
  transition: all 0.5s ease 0.25s;
}

.emotion-4:hover {
  visibility: visible;
  opacity: 1;
}

.emotion-3:before {
  content: ' ';
  position: absolute;
  top: -6px;
  right: 14px;
  border-bottom: 6px solid #ffffff;
  border-left: 6px solid transparent;
  border-right: 6px solid transparent;
}

<div
  className="emotion-7"
>
  <FaceLebalingMessage
    error=""
    jobId=""
    previousJobId=""
    projectId="76917058-b147-4556-987a-0a0f11e46d9b"
    setPreviousJobId={[Function]}
  />
  <span>
    Once a name has been added to a face, training can begin. Names can continue to be edited as needed.
  </span>
  <div
    className="emotion-0"
  />
  <div
    className="emotion-6"
  >
    <button
      aria-disabled={true}
      className="emotion-1"
      onClick={[Function]}
      type="button"
    >
      Train & Apply
    </button>
    <div
      className="emotion-5"
    >
      <button
        aria-details="TrainingHelp"
        aria-label="Training Help"
        className="emotion-2"
        type="button"
      >
        <svg
          data-file-name="help.svg"
          height={20}
        />
      </button>
      <div
        className="emotion-4"
        id="TrainingHelp"
        role="tooltip"
      >
        <div
          className="emotion-3"
        />
        There are new labels. Click to train and apply.
      </div>
    </div>
  </div>
</div>
`;
